### PR TITLE
Guide user and prevent empty searches

### PR DESCRIPTION
**Pull request by AbhiramTadepalli**
## Overview

#215

Grey out search button text and show tooltip if user tries to empty search

## What Changed

Search button text is not white by default, and becomes white once the user selects at least 1 valid search term from autocomplete
Removed the alert from index.tsx that used to display above the search bar when the user tried to empty search
Transferred the text into a tooltip that pops up above the search button if the user clicks enter or the button without any search terms

### CODE REVIEW

**Comment by AbhiramTadepalli** [comment]
Should the text be "Select a course or professor before searching" instead of "Select an option before searching"

File: src/pages/index.tsx [hunk 1]
```diff
@@ -14,11 +13,8 @@ import searchQueryLabel from '../modules/searchQueryLabel/searchQueryLabel';
  * Returns the home page with Nebula Branding, waved background, and SearchBar Components
  */
 const Home: NextPage = () => {
-  const [errorMessage, setErrorMessage] = useState(false);
   function searchOptionsChange(chosenOptions: SearchQuery[]) {
-    if (chosenOptions.length) {
-      setErrorMessage(false);
-    }
+    // does nothing on change
   }
```
Comment (TyHil) [inline src/pages/index.tsx:18]: can you remove this function if it does nothing

File: src/components/common/SearchBar/searchBar.tsx [hunk 2]
```diff
@@ -332,15 +334,30 @@ const SearchBar = ({
...
-        Search
-      </Button>
+        <Button
+          variant="contained"
+          disableElevation
+          size="large"
+          className={
+            value.length == 0
+              ? 'shrink-0 normal-case bg-royal hover:bg-royalDark text-cornflower-100'
+              : 'shrink-0 normal-case bg-royal hover:bg-royalDark'
```
Comment (TyHil) [inline src/components/common/SearchBar/searchBar.tsx:354]: can this be:

`'shrink-0 normal-case bg-royal hover:bg-royalDark' + (value.length == 0 ? ' text-cornflower-100' : '')`

so it's more obvious what's changing

File: src/components/common/SearchBar/searchBar.tsx [hunk 3]
```diff
@@ -192,6 +193,7 @@ const SearchBar = ({
 
   //update parent and queries
   function onSelect_internal(newValue: SearchQuery[]) {
+    setErrorTooltip(!newValue.length); //Check if tooltip needs to be displayed
```
Comment (TyHil) [inline src/components/common/SearchBar/searchBar.tsx:196]: the tool tip never goes away on the homepage, not sure why, seems this should do it.

**Review by TyHil — CHANGES_REQUESTED** [review]
The search bar still lets you search with a blank field on the dashboard, should make it not allow you like on the homepage?

Otherwise, left some comments

**Comment by TyHil** [comment]
> Should the text be "Select a course or professor before searching" instead of "Select an option before searching"?

That sounds good, fine with either

**Comment by AbhiramTadepalli** [comment]
> The search bar still lets you search with a blank field on the dashboard, should make it not allow you like on the homepage?
> 
> Otherwise, left some comments

Fixed this and everything else, can you check?

**Comment by AbhiramTadepalli** [comment]
The search button is a _little_ shorter than the searchbar, is that intentional?

**Comment by TyHil** [comment]
Looks all good to me!

> The search button is a _little_ shorter than the searchbar, is that intentional?

Unintentional, just mui default button height

**Review by TyHil — APPROVED** [review]

**Comment by AbhiramTadepalli** [comment]
> Unintentional, just mui default button height

should I change?

**Pull request merged**